### PR TITLE
Discovered classes are now sorted - fixing #1180

### PR DESCRIPTION
**Pull request by kretes**

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

File: main/src/main/scala/sbt/Defaults.scala [hunk 1]
```diff
@@ -739,7 +739,7 @@ object Defaults extends BuildCommon {
   def mainRunMainTask = runMain <<= runMainTask(fullClasspath in Runtime, runner in run)
 
   def discoverMainClasses(analysis: inc.Analysis): Seq[String] =
-    Discovery.applications(Tests.allDefs(analysis)) collect { case (definition, discovered) if (discovered.hasMain) => definition.name }
+    Discovery.applications(Tests.allDefs(analysis)) collect { case (definition, discovered) if discovered.hasMain => definition.name } sorted
```
Comment (jsuereth) [inline main/src/main/scala/sbt/Defaults.scala]: we try to (now) avoid postfix notation.

**Comment by jsuereth** [comment]
minor style comment, otherwise LGTM

File: main/src/main/scala/sbt/Defaults.scala (hunk 1 above)
Comment (kretes) [inline main/src/main/scala/sbt/Defaults.scala]: Ok, I can refactor that.
22 paź 2014 15:27 "Josh Suereth" notifications@github.com napisał(a):

**Comment by jsuereth** [comment]
This appears to have borked a test: https://travis-ci.org/sbt/sbt/jobs/38641891

**Comment by kretes** [comment]
I guess it is because it branched out of 0.13 in a bad moment. current build https://travis-ci.org/sbt/sbt/jobs/38622944 is failing as well.

**Comment by jsuereth** [comment]
you're right.  master should be good now, was a merge issue.

**Comment by jsuereth** [comment]
Ping, this can be rebased against master, now that master is stable.

Thanks again for the patches!

**Comment by kretes** [comment]
yep, will do that soon

**Comment by kretes** [comment]
Will open new for 0.13.7

**Pull request closed without merging**